extension/observer/dockerobserver/extension_test.go

```diff
@@ -0,0 +1,302 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dockerobserver
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"path"
+	"syscall"
+	"testing"
+
+	dtypes "github.com/docker/docker/api/types"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/config/configtest"
+	"go.uber.org/zap"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/extension/observer"
+)
+
+func containerJSON(t *testing.T) dtypes.ContainerJSON {
+	containerRaw, err := ioutil.ReadFile(path.Join(".", "testdata", "container.json"))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	var container dtypes.ContainerJSON
+	err = json.Unmarshal(containerRaw, &container)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return container
+}
+
+func TestPortTypeToProtocol(t *testing.T) {
+	tests := []struct {
+		name     string
+		portType uint32
+		want     observer.Transport
+	}{
+		{
+			name:     "tcp",
+			portType: syscall.SOCK_STREAM,
+			want:     observer.ProtocolTCP,
+		},
+		{
+			name:     "udp",
+			portType: syscall.SOCK_DGRAM,
+			want:     observer.ProtocolUDP,
+		},
+		{
+			name:     "unsupported",
+			portType: 999,
+			want:     observer.ProtocolUnknown,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := portProtoToTransport(tt.name); got != tt.want {
+				t.Errorf("portTypeToProtocol() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestCollectEndpointsDefaultConfig(t *testing.T) {
+	factory := NewFactory()
+	ext, err := newObserver(zap.NewNop(), factory.CreateDefaultConfig().(*Config))
+	require.NoError(t, err)
+	require.NotNil(t, ext)
+
+	obvs := ext.(*dockerObserver)
```

Comment:
potentially helps w/ clarity
```suggestion
	obvs, ok := ext.(*dockerObserver)
	require.True(t, ok)
```